### PR TITLE
Fizz buzz challenge

### PR DESCRIPTION
**Pull request by lanamusienko**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by MUSABKAYMAK — COMMENTED** [review]
Overall very good. No implementation problem. Few small changes are needed to complete the checklist.

File: solutions/fizzbuzz_challenge.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - fizzbuzz: take integer n and prints numbers from 1 to n
+        with the rules.
+
+Created: 01/04/2025
+Author: Svitlana Musiienko
+
+"""
+
+
+def fizzbuzz(n):
```
[the remark this answers is not in the document]
Comment (lanamusienko) [reply solutions/fizzbuzz_challenge.py:14]: The  file name is corrected and now match the function name.
The type annotations of argument and return value added, thank you for suggestion.

File: solutions/fizzbuzz_challenge.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    Fizzbuz function takes an integer n as input and prints the numbers
+    from 1 to n with the rules:
+        - if a number is divisible by 3, print 'Fizz' instead of the number;
+        - if a number is divisible by 5, print 'Buzz' instead of the number;
+        - if a number is divisible by both 3 and 5, print 'FizzBuzz';
+        - otherwise, print the number itself.
+
+    Parameters: integer number (n)
+
+    Returns: print numbers from 1 to n based on given rules.
```
[the remark this answers is not in the document]
Comment (lanamusienko) [reply solutions/fizzbuzz_challenge.py:25]: Yue are correct, I changed the the return value for return list of numbers from 1 to n where numbers divisible by 3 and 5 are replaced with fizz, buzz and fizzbuzz

File: solutions/tests/test_fizzbuzz_challenge.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            11,
+            "Fizz",
+            13,
+            14,
+            "FizzBuzz",
+        ]
+        self.assertEqual(actual, expected)
+
+    def test_other(self):
+        """It should return numbers till 2"""
```
[the remark this answers is not in the document]
Comment (lanamusienko) [reply solutions/tests/test_fizzbuzz_challenge.py:51]: Corrected to more general description """It should check the numbers which are not devisable by 3 or 5"""

File: solutions/tests/test_fizzbuzz_challenge.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import unittest
+
+from ..fizzbuzz_challenge import fizzbuzz
+
+
+class TestFizzBuzz(unittest.TestCase):
+    """Test the fizzbuzz function"""
+
+    def test_fizz(self):
+        """It should print "Fizz" instead 3 and 6 and return other numbers till 6"""
```
[the remark this answers is not in the document]
Comment (lanamusienko) [reply solutions/tests/test_fizzbuzz_challenge.py:17]: You are right, I made changes through the entire test file docstring and made description more general and clear.

File: solutions/tests/test_fizzbuzz_challenge.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        actual = fizzbuzz(6)
+        expected = [1, 2, "Fizz", 4, "Buzz", "Fizz"]
+        self.assertEqual(actual, expected)
+
+    def test_buzz(self):
+        """It should print "Buzz" instead 5 and 10 and return other numbers till 10"""
```
[the remark this answers is not in the document]
Comment (lanamusienko) [reply solutions/tests/test_fizzbuzz_challenge.py:23]: I've made changes for general style of docstring for the test file fizzbuzz_chalange.

**Comment by lanamusienko** [comment]
@MUSABKAYMAK thank you for all comments, they all are important and making sense, really improving my docstring. Please review changes.

**Review by MUSABKAYMAK — APPROVED** [review]
